enhancements/installer/gcp-ipi-shared-vpc.md

```diff
@@ -0,0 +1,282 @@
+
+#### installconfig.platform.gcp.createFirewallRules
+
+A new parameter will be added to the install-config to enable toggling of the
+creation and management of firewall rules. When set to disabled, the installer
+will skip creation of firewall rules. The default is enabled in which firewall
+rules will be created as before. This toggle does not modify the behavior of
+the cluster-ingress-operator, which is to create firewall rules if permissions
+allow.
+
+### Implementation Details/Notes/Constraints
+
+#### VPC, Network, and Subnetworks
+
+Shared VPC network and subnet(s) are to be created and configured in advance.
+They can then be configured in the install config using the following
+parameters as enabled by prior work. A new parameter (networkProjectID) will be
+added to specify in which project they exist. This parameter will be consumed
+by the Terraform templates as well as cluster manifests in order to configure
+the cluster to use these resources.
+
+- installconfig.platform.gcp.computeSubnet
+- installconfig.platform.gcp.controlPlaneSubnet
+- installconfig.platform.gcp.network
+
+#### Service Accounts
+
+Service Account(s) are to be created and configured in advance. These accounts
+must have additional permissions granted in the Host Project. The specific
+permissions depend on the desired level of automation. The install config's
+credentialsMode parameter must be set to passthrough so the cluster uses these
+accounts.
+
+#### Firewall Rules
+
+Firewall Rules can either be created and configured in advance, or the service
+accounts above must have sufficient permissions granted to manage firewall
+rules in the Host Project. User specified tags will be added to the instances
+to enable cluster specific rules. A new install config parameter will be added
+to enable specifying if the Terraform templates should create firewall rules,
+or not.
+
+In order for the firewall rules to be created by the installer, the service
+account will require the following access in the project where the firewall
+rules are to be managed (i.e. networkProjectID)
+
+- Compute Network Admin
+- Compute Security Admin
+
+#### DNS Zones and Records
+
+DNS can either be created and configured in advance, or the service accounts
+above must have sufficient permissions to create and manage DNS resources in
+the Host Project.
+
+The ability to use pre-existing (aka external) DNS solutions is outside the
+scope of this enhancement. There are other efforts to provide this capability.
+
+New install config parameters will be added to specify the public and/or
+private DNS zones to use when managing cluster records. The default will be an
+empty value, which will result in the private zone and public/private records
+being managed in the service account as before.
+
+In order for the DNS Zones and Records to be managed by the installer, the
+service account will require the following access in the in the project(s)
```

Comment:
Do you mean the ingress operator will need to have sufficient permissions as it will be the one who will fetch the values from the install-config yaml and create the resources for you in the Host Project and gcloud ? or will the installer create the dns zones and records using CCO in the host project ? cc @Miciah

Reply:
The installer will create the private zone and add some records to both the private and public zone. It will need sufficient permissions to do so. The cluster-ingress-operator will create some records in both zones and will need permissions to do so. The workflow for this will require passthrough credentials mode to enable the user to provide a service account with pre-configured permissions to do both.